std/btree.asm

```diff
@@ -98,19 +99,36 @@ mod internal {
     };
     let<K, V> split_leaf: (K, V)[] -> InsertResult<K, V> = |items| {
         let split = (std::array::len(items) - 1) / 2;
-        let left = sub_array(items, 0, split);
-        let right = sub_array(items, split + 1, std::array::len(items) - split - 1);
-        InsertResult::Split(items[split], BTree::Leaf(left), BTree::Leaf(right))
+        let (left, center, right) = array_split_pivot(items, split);
+        InsertResult::Split(center, BTree::Leaf(left), BTree::Leaf(right))
+    };
+    /// Splits an array into left and right part. The element at index i is
+    /// not part of either and returned separately.
+    let<T> array_split_pivot: T[], int -> (T[], T, T[]) = |arr, i| {
```

Comment:
I think my previous comment still applies: `i` should always equal `super::max_items / 2`, so I would remove that argument and add an `assert(std::array::len(arr) == super::max_items + 1, || "Unexpected number of items");`

Reply:
I think it's better to keep a generic function generic instead of pulling invariants from the outside into the helper function